src/qat/purr/compiler/optimisation_passes.py

```diff
@@ -0,0 +1,88 @@
+from numbers import Number
+from typing import Dict, List
+
+from qat.ir.pass_base import TransformPass
+from qat.ir.result_base import ResultManager
+from qat.purr.compiler.builders import InstructionBuilder
+from qat.purr.compiler.devices import PulseChannel
+from qat.purr.compiler.instructions import (
+    AcquireMode,
+    CustomPulse,
+    Instruction,
+    PhaseReset,
+    PhaseShift,
+    PostProcessing,
+    PostProcessType,
+    ProcessAxis,
+    Pulse,
+)
+
+
+class PhaseOptimisation(TransformPass):
+    """
+    Extracted from QuantumExecutionEngine.optimize()
```

Comment:
Does this mean that, from the moment these passes are adapted in the code, we can safely omit the `optimize` method in `QuantumExecutionEngine`?

Reply:
correct ^^, such methods can be safely omitted